spec.md

```diff
@@ -230,6 +245,39 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Public Modules
+
+A public module is accessible by `Circuit`s other than the `Circuit` it is
+defined in. Unless otherwise specified, all modules inside of a `Circuit` are
+private and cannot be accessed by any other `Circuit`.
+
+A public module has fixed ports.
+
+A module can be marked as public like so:
+
+```firrtl
+circuit FooCircuit :
+   module Foo is public :
+      input foo: UInt
+      output bar: UInt
+      ; ...
+```
+
+A public module can be instantiated like so:
```

Comment:
```suggestion
A public module can be instantiated (in a different Circuit) like so:
```

Not sure if we want to mention that it will be instantiated in *this* Circuit as normal / in the same way it is instantiated if it were private?